### PR TITLE
add tests for bonds

### PR DESCRIPTION
**Pull request by jdaaph**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,123 @@
+# Byte-compiled / optimized / DLL files
```
Comment (bdice) [inline .gitignore]: @jdaaph Could you choose a slightly more conservative `.gitignore`? Maybe compare what freud and signac do, for reference. At least cut out all the stuff that is irrelevant (e.g. Flask, Django, celery, Sage, Rope, Scrapy, ...). Keep things like packaging and `.ipynb_checkpoints`.

File: pythia/bonds.py [hunk 2]
```diff
@@ -100,6 +100,9 @@ def neighborhood_distance_sorted(box, positions, neighbors, rmax_guess=2.):
 
 
 def get_neighborhood_angle_matrix(box, positions, neighbors, rmax_guess=2.):
+    """Construct a matrix of pairwise angles between `|r_j - r_i|` and `|r_k - r_i|`
+    for all neighbors j and k(==j) of each particle i.
```
Comment (bdice) [inline pythia/bonds.py]: Do you mean `k (k != j)`?

File: pythia/voronoi.py [hunk 3]
```diff
@@ -65,8 +65,11 @@ def angle_histogram(box, positions, bins, buffer_distance=None, area_weight_mode
     :param area_weight_mode: Whether the weight for each pair of faces should be the sum ('sum') or product ('product') of the face areas
     """  # noqa E501
 
+    # The buffer distance is used to produce image for the periodic boundary condition,
+    # to avoid corner cases like particles are only in one quadrant, we need to take the max
+    # value of the box dimension as the replication buffer distance.
     if buffer_distance is None:
-        buffer_distance = min(box.Lx, box.Ly, box.Lz)*.5
+        buffer_distance = max(box.Lx, box.Ly, box.Lz)
```
Comment (bdice) [inline pythia/voronoi.py]: Note for @klarh - it seems that the previous buffer distance was insufficient in sparse systems like the test system that @jdaaph created. It seems that one full periodic buffer in each direction is needed.

File: pythia/bonds.py (hunk 2 above)
Comment (jdaaph) [reply pythia/bonds.py]: actually j can be equal to k so i just deleted it.

**Comment by jdaaph** [comment]
This is no longer WIP ready for merge.

**Comment by klarh** [comment]
The `get_neighborhood_{distance,angle}_matrix` functions were intended to be only internally-used; I prefixed their name with an underscore.

**Pull request merged**